SCloudRandomizer/MainViewController.m

```diff
@@ -310,86 +309,98 @@ - (void)getTrackInfo:(NSDictionary *)track shouldPlay:(BOOL)play
     NSString *streamURL = [track objectForKey:@"stream_url"];
     NSLog(@"The streamURL is: %@", streamURL);
     
+    // Get new track if stream URL is nil
```

Comment:
I am a little confused as to what this while block is doing and why we have it this way but lets discuss that in person